openmdao/surrogate_models/kriging.py

```diff
@@ -150,6 +156,14 @@ def _calculate_reduced_likelihood_params(self, thetas=None):
         params['U'] = U
         params['Vh'] = Vh
 
+        # Newly added for EGOLF
+        one = np.ones([self.n_samples,1])
+        R_inv = Vh.T.dot(np.dot(np.diag(inv_factors),U.T)) #Use einsum in the future release for efficiency
```

Comment:
IF R^{-1} is explicitly needed, use `Vh.T.dot(np.einsum('i,ij->ij', inv_factors, U.T))`